src/lint/rules/subject-capitalize-first.ts

```diff
@@ -0,0 +1,15 @@
+import { createRule, RuleFactoryCheckResults } from "./factory";
+import { Commit } from "@commitlint/parse";
+
+export default createRule({
+  name: "subject-capitalize-first",
+  score: 5,
+  check: (commit: Commit): RuleFactoryCheckResults => {
+    const codePoint = commit.header.codePointAt(0);
```

Comment:
We may use Regex with unicode character class (`/^\p{Lu}/u`) to simplify the logic.